src/pub_verif_token.ts

```diff
@@ -139,55 +153,59 @@ export class Issuer {
 
 export class Client {
     static readonly TYPE = BLIND_RSA;
+
     private finData?: {
-        publicKeyIssuer: CryptoKey;
+        pkIssuer: CryptoKey;
         tokenInput: Uint8Array;
-        tokenPayload: TokenPayload;
-        tokenRequest: TokenRequest;
+        authInput: AuthenticatorInput;
         inv: Uint8Array;
     };
 
-    async createTokenRequest(privToken: PrivateToken): Promise<TokenRequest> {
+    async createTokenRequest(
+        tokChl: TokenChallenge,
+        issuerPublicKey: Uint8Array,
+    ): Promise<TokenRequest> {
         // https://datatracker.ietf.org/doc/html/draft-ietf-privacypass-protocol-11#section-6.1
         const nonce = crypto.getRandomValues(new Uint8Array(32));
-        const context = new Uint8Array(
-            await crypto.subtle.digest('SHA-256', privToken.challengeSerialized),
+        const context = new Uint8Array(await crypto.subtle.digest('SHA-256', tokChl.serialize()));
+
+        const tokenKeyId = await getTokenKeyID(issuerPublicKey);
+        const authInput = new AuthenticatorInput(
+            Client.TYPE,
+            Client.TYPE.value,
+            nonce,
+            context,
+            tokenKeyId,
         );
-        const keyId = new Uint8Array(await crypto.subtle.digest('SHA-256', privToken.tokenKey));
-        const tokenPayload = new TokenPayload(Client.TYPE, nonce, context, keyId);
-        const tokenInput = tokenPayload.serialize();
+        const tokenInput = authInput.serialize();
 
         const blindRSA = SUITES.SHA384.PSS.Deterministic();
-        const spkiEncoded = convertRSASSAPSSToEnc(privToken.tokenKey);
-        const publicKeyIssuer = await crypto.subtle.importKey(
-            'spki',
-            spkiEncoded,
-            { name: 'RSA-PSS', hash: 'SHA-384' },
-            true,
-            ['verify'],
-        );
+        const pkIssuer = await getCryptoKey(issuerPublicKey);
 
-        const { blindedMsg, inv } = await blindRSA.blind(publicKeyIssuer, tokenInput);
-        const tokenKeyId = keyId[keyId.length - 1];
-        const tokenRequest = new TokenRequest(tokenKeyId, blindedMsg);
-        this.finData = { tokenInput, tokenPayload, inv, tokenRequest, publicKeyIssuer };
+        const { blindedMsg, inv } = await blindRSA.blind(pkIssuer, tokenInput);
+        const trucatedTokenKeyId = tokenKeyId[tokenKeyId.length - 1];
```

Comment:
Add a comment about why this is the latest element.

nit: typo missing `n`
```suggestion
        // truncatedTokenKeyId is the least significant byte of the token_key_id. With Uint8Array, this is the last element.
        const truncatedTokenKeyId = tokenKeyId[tokenKeyId.length - 1];
```